RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.cc

```diff
@@ -0,0 +1,583 @@
+  int flip = pixelFlipper(det);  // 1=not flip, -1=flip
+
+  for (int i = 0; i < cluster.size(); i++) {
+    SiPixelCluster::Pixel pix = cluster.pixel(i);
+    std::pair<int, int> pixInter = local2Pixel(inter.x(), inter.y(), det);
+    int nx = pix.x - pixInter.first;
+    int ny = pix.y - pixInter.second;
+    nx = flip * nx;
+
+    if (abs(nx) < jetDimX / 2 && abs(ny) < jetDimY / 2) {
+      nx = nx + jetDimX / 2;
+      ny = ny + jetDimY / 2;
+      //14000 = normalization of ACD counts used in the NN
+      input_tensors[2].second.tensor<float, 4>()(0, nx, ny, layer - 1) += (pix.adc) / (14000.f);
+    }
+  }
+}
+
+std::pair<double[DeepCoreSeedGenerator::jetDimX][DeepCoreSeedGenerator::jetDimY][DeepCoreSeedGenerator::Nover]
+                [DeepCoreSeedGenerator::Npar],
+          double[DeepCoreSeedGenerator::jetDimX][DeepCoreSeedGenerator::jetDimY][DeepCoreSeedGenerator::Nover]>
+DeepCoreSeedGenerator::SeedEvaluation(tensorflow::NamedTensorList input_tensors,
+                                      std::vector<std::string> output_names) {
+  std::vector<tensorflow::Tensor> outputs;
+  tensorflow::run(session_, input_tensors, output_names, &outputs);
+  auto matrix_output_par = outputs.at(0).tensor<float, 5>();
+  auto matrix_output_prob = outputs.at(1).tensor<float, 5>();
+
+  std::pair<double[jetDimX][jetDimY][Nover][Npar], double[jetDimX][jetDimY][Nover]> output_combined;
+
+  for (int x = 0; x < jetDimX; x++) {
+    for (int y = 0; y < jetDimY; y++) {
+      for (int trk = 0; trk < Nover; trk++) {
+        output_combined.second[x][y][trk] =
+            matrix_output_prob(0, x, y, trk, 0);  //outputs.at(1).matrix<double>()(0,x,y,trk);
+
+        for (int p = 0; p < Npar; p++) {
+          output_combined.first[x][y][trk][p] =
+              matrix_output_par(0, x, y, trk, p);  //outputs.at(0).matrix<double>()(0,x,y,trk,p);
+        }
+      }
+    }
+  }
+  return output_combined;
+}
+
+const GeomDet* DeepCoreSeedGenerator::DetectorSelector(int llay,
+                                                       const reco::Candidate& jet,
+                                                       GlobalVector jetDir,
+                                                       const reco::Vertex& jetVertex,
+                                                       const TrackerTopology* const tTopo,
+                                                       const edmNew::DetSetVector<SiPixelCluster>& clusters) {
+  struct trkNumCompare {
+    bool operator()(std::pair<int, const GeomDet*> x, std::pair<int, const GeomDet*> y) const {
+      return x.first > y.first;
+    }
+  };
+  std::set<std::pair<int, const GeomDet*>, trkNumCompare> track4detSet;
+
+  double minDist = 0.0;
+  GeomDet* output = (GeomDet*)nullptr;
+  for (const auto& detset : clusters) {
+    const GeomDet* det = geometry_->idToDet(detset.id());
+    auto aClusterID = detset.id();
+    if (DetId(aClusterID).subdetId() != 1)
```

Comment:
hi @vberta 

this comparison would be better down before the idToDet call.

Reply:
ok